controller/service.go

```diff
@@ -128,6 +144,13 @@ func (c *controller) clearServiceState(key string, svc *v1.Service) {
 }
 
 func (c *controller) allocateIP(key string, svc *v1.Service) (net.IP, error) {
+	clusterIP := net.ParseIP(svc.Spec.ClusterIP)
+	if clusterIP == nil {
+		// (we should never get here)
```

Comment:
Can you add "because the caller ensured that Spec.ClusterIP != nil" ? So that in 2 years we remember why it should never get here :)